library/core/src/ptr/non_null.rs

```diff
@@ -502,6 +502,8 @@ impl<T: ?Sized> NonNull<T> {
     #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
     #[stable(feature = "non_null_convenience", since = "1.80.0")]
     #[rustc_const_stable(feature = "non_null_convenience", since = "1.80.0")]
+    #[kani::requires(kani::mem::same_allocation(self.as_ptr() as *const(), self.as_ptr().byte_offset(count) as *const()))]
```

Comment:
Is the cast to `*const()` necessary?

Reply:
> Is the cast to `*const()` necessary?

Good point! The reason I did the const cast here is because previously `same_allocation()` is not support `Size?` input, and I got a workaround suggestion from https://github.com/model-checking/verify-rust-std/discussions/141